### PR TITLE
update Code of Conduct to version 2

### PR DESCRIPTION
**Pull request by zkamvar**
The previous version of the Code of Conduct (CoC) contained language that was too specific to contributions to code repositories. Moreover, there was no clear definition of what the consequences for those who violated the CoC were beyond the scope of code. This version of the CoC fills the gaps a lot better.

I would like input from the community about these changes.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,27 +1,90 @@
 ---
-title: RECON Contributor Code of Conduct
+title: RECON Code of Conduct
 ---
 
-As contributors and maintainers of this project, we pledge to respect all people who 
-contribute through reporting issues, posting feature requests, updating documentation,
-submitting pull requests or patches, and other activities.
```
Comment (AmmerB) [inline CODE_OF_CONDUCT.md:7]: - Contribute to RECON training activities or deployments

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,27 +1,90 @@
...
+
+Community Impact Guidelines were inspired by [Mozilla's code of conduct enforcement ladder](https://github.com/mozilla/diversity).
+
+[homepage]: https://www.contributor-covenant.org
+
+For answers to common questions about this code of conduct, see the FAQ at
+https://www.contributor-covenant.org/faq. Translations are available at https://www.contributor-covenant.org/translations.
 
-Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by 
-opening an issue or contacting one or more of the project maintainers.
```
Comment (AmmerB) [inline CODE_OF_CONDUCT.md:23]: Such issues will discussed by the RECON advisory board and appropriate actions will be taken after careful and deliberate review of such complaints.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,27 +1,90 @@
...
+
+**Community Impact**: A serious violation of community standards, including sustained inappropriate behavior.
+
+**Consequence**: A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, is allowed during this period. Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+
+**Community Impact**: Demonstrating a pattern of violation of community standards, including sustained inappropriate behavior,  harassment of an individual, or aggression toward or disparagement of classes of individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction within the RECON community.
```
Comment (AmmerB) [inline CODE_OF_CONDUCT.md:77]: If required legal action can be taken, or such action can be reported to the appropriate authorities

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (zkamvar) [reply CODE_OF_CONDUCT.md:7]: You're commenting on deleted text here. I believe this is covered in the "Scope" paragraph below: 

> This Code of Conduct applies within all RECON community spaces

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (zkamvar) [reply CODE_OF_CONDUCT.md:23]: This is on removed text, but is covered in the Enforcement section:

> All complaints will be reviewed and investigated promptly and fairly.
>
> All community leaders are obligated to respect the privacy and security of the reporter of any incident.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (zkamvar) [reply CODE_OF_CONDUCT.md:77]: I'm not sure how this sentence works. How about this:

```suggestion
**Consequence**: A permanent ban from any sort of public interaction within the RECON community. 
Any action requiring legal intervention will be reported to the appropriate authorities.
```

Comment (thibautjombart) [reply CODE_OF_CONDUCT.md:77]: I think the comment on legal intervention should be made before detailing the different grades of response, as it applies at least in theory to all of them.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -1,27 +1,90 @@
...
 
-We are committed to making participation in this project a harassment-free experience for
-everyone, regardless of level of experience, gender, gender identity and expression,
-sexual orientation, disability, personal appearance, body size, race, ethnicity, age, or religion.
+## Our Pledge
 
-Examples of unacceptable behavior by participants include the use of sexual language or
-imagery, derogatory comments or personal attacks, trolling, public or private harassment,
-insults, or other unprofessional conduct.
+We as members, contributors, and leaders pledge to make participation in the RECON community a harassment-free experience for everyone, regardless of age, body size, visible or invisible disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
```
Comment (scolombe) [inline CODE_OF_CONDUCT.md]:
```suggestion
We as members, contributors, and leaders pledge to make participation in the RECON community a harassment-free experience for everyone, regardless of age, body size, visible or invisible disability, ethnicity, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, actual or perceived sexual identity and orientation, gender identity and expression, or sex characteristics. 
```

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -1,27 +1,90 @@
...
+Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, and will communicate reasons for moderation decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all RECON community spaces, and also applies when an individual is officially representing the RECON community in public spaces. Examples of representing our community include using an official e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the community leaders responsible for enforcement by emailing [repidemicsconsortium@gmail.com](mailto:repidemicsconsortium@gmail.com?subject=CoCEnforcement) or contacting one of the [Executive Board](https://www.repidemicsconsortium.org/people#executive-board) members via the [RECON slack channel](https://reconhub.slack.com) or private email. All complaints will be reviewed and investigated promptly and fairly.
+
```
Comment (scolombe) [inline CODE_OF_CONDUCT.md]: Use a separate email address and to be discussed whether the executive board is the best contact to receive first hand complaints.

**Comment by cwhittaker1000** [comment]
Have just pushed a new commit providing updates to the Code of Conduct. These updates are:
1. Incorporating in @scolombe's additions and extensions to the protected characteristics section.
2. Incorporating in detail about the new reporting guidelines being put in place (specifically, the new google form, the new email with automatic onwards forwarding to the CoC committee etc).
3. Attribution to rOpenSci, recognising their current reporting guidelines as the inspirations for the system we have put in place here.
Aim is to get feedback on this from the board and pass at the next RECON board meeting (early January).

**Pull request merged**